src/plugins/intel_npu/tests/unit/npuw/serialization.cpp

```diff
@@ -0,0 +1,279 @@
+    using namespace ov::npuw::s11n;
+
+    std::vector<uint8_t> data {0, 1, 2, 3};
+    ov::Tensor var(ov::element::u8, ov::Shape({2, 2}), data.data());
+    ov::Tensor res;
+
+    std::stringstream ss;
+
+    write(ss, var);
+    read(ss, res);
+
+    EXPECT_EQ(res.get_element_type(), ov::element::u8);
+    EXPECT_EQ(res.get_shape(), ov::Shape({2, 2}));
+
+    std::vector<uint8_t> data_res(4, 0);
+    std::memcpy(data_res.data(), res.data(), 4);
+
+    EXPECT_EQ(data, data_res);
+}
+
+TEST(SerializationTest, OVTypes_Spatial) {
+    using namespace ov::npuw::s11n;
+
+    ov::npuw::compiled::Spatial var;
+    var.params = {{0, 1}, {2, 3}};
+    var.range = 3;
+    var.nway = 5;
+    var.out_dim = 1;
+    var.nway_iters = 10;
+    var.tail_size = 3;
+
+    ov::npuw::compiled::Spatial res;
+
+    std::stringstream ss;
+
+    write(ss, var);
+    read(ss, res);
+
+    EXPECT_EQ(var.params[0].idx, res.params[0].idx);
+    EXPECT_EQ(var.params[0].dim, res.params[0].dim);
+    EXPECT_EQ(var.params[1].idx, res.params[1].idx);
+    EXPECT_EQ(var.params[1].dim, res.params[1].dim);
+    EXPECT_EQ(var.range, res.range);
+    EXPECT_EQ(var.nway, res.nway);
+    EXPECT_EQ(var.out_dim, res.out_dim);
+    EXPECT_EQ(var.nway_iters, res.nway_iters);
+    EXPECT_EQ(var.tail_size, res.tail_size);
+}
+
+TEST(SerializationTest, OVTypes_Config) {
+    using namespace ov::npuw::s11n;
+
+    auto options_desc(std::make_shared<::intel_npu::OptionsDesc>());
+    options_desc->add<::intel_npu::NPUW_LLM_BATCH_DIM>();
+    ::intel_npu::Config var(options_desc);
+    ::intel_npu::Config res(options_desc);
+
+    std::map<std::string, std::string> tmp;
+    tmp["NPUW_LLM_BATCH_DIM"] = "42";
+    var.update(tmp);
+
+    std::stringstream ss;
+
+    write(ss, var);
+    read(ss, res);
+
+    EXPECT_EQ(res.get<::intel_npu::NPUW_LLM_BATCH_DIM>(), 42);
+}
+
+TEST(SerializationTest, OVTypes_Any) {
+    using namespace ov::npuw::s11n;
+
+    std::vector<ov::Any> var;
+    var.push_back(42);
+    var.push_back("42");
+    var.push_back(float(3.14));
+    var.push_back(true);
+    std::vector<ov::Any> res;
+    res.resize(var.size());
+
+    std::stringstream ss;
+
+    for (std::size_t i = 0; i < var.size(); ++i) {
+        write_any(ss, var[i]);
+        read_any(ss, res[i]);
+        EXPECT_EQ(var[i], res[i]);
+    }
+}
+
+TEST(SerializationTest, BasicTypes_array) {
+    using namespace ov::npuw::s11n;
+
+    IndicatorType res;
+
+    std::stringstream ss;
+
+    write(ss, NPUW_SERIALIZATION_INDICATOR);
+    read(ss, res);
+
+    EXPECT_EQ(NPUW_SERIALIZATION_INDICATOR, res);
+}
+
+TEST(SerializationTest, BasicTypes_pair) {
+    using namespace ov::npuw::s11n;
+
+    std::pair<int, float> var{42, 3.14};
+    std::pair<int, float> res;
+
+    std::stringstream ss;
+
+    write(ss, var);
+    read(ss, res);
+
+    EXPECT_EQ(var, res);
+}
+
+TEST(SerializationTest, BasicTypes_vector) {
+    using namespace ov::npuw::s11n;
+
+    std::vector<int> var{1, 2, 3, 45};
```

Comment:
Will test for more complex vectors work? Like `std::vector<ov::Any>`?

Reply:
Not with `Any`. Other types which are supported with `write()` will